### ISSUE TITLE
Implement Feature Toggles

### ISSUE THREAD

**Issue by Odonno** [Odonno/Modern-Gitter#37]
Add Feature Toggles in the project to make changes without breaking the entire application until the new version.

**Comment by gep13** [comment]
Out of interest, what is going to control these feature toggles?  A new Settings screen, or something else?

**Comment by Odonno** [comment]
Yeah, for example, but it can be everything : 

* a toggle inside the app (settings page),
* the code himself (like a hardcoded boolean, a calculation, a store value, ...),
* or even an admin settings available on a dedicated website

**Comment by gep13** [comment]
Cool.  For the last one, you might want to incorporate something similar to what is in Gitter itself, i.e. an option to subscribe to the "Next" Version of Gitter, where all new features are enabled by default, or turned on/off as required.

**Comment by Odonno** [comment]
Yep, it is exactly that. I don't know what they are using exactly (staging environment or feature toggle), but it is in the way. Feature Toggles allow you to define precisely, for each feature, if it is enabled or disabled, using single or multiple conditions.

**Comment by gep13** [comment]
Not sure on exactly what you were planning to use for this, but you might want to take a look at this:

https://github.com/opentable/hobknob

I remembered I had read about it, and thought I would share.

**Comment by Odonno** [comment]
i did not know that. It seems great but I prefer another package (https://github.com/jason-roberts/FeatureToggle). 

Compared to what I saw of hobknob, FeatureToggle provide XAML compatibility (like converters). Although it is as simple to integrate with MVVMLight. For an example, I have started another project with MVVMLight and FeatureToggle here : https://github.com/Odonno/C-est-pas-faux

**Comment by gep13** [comment]
Yes, I am familiar with that Feature Toggle project as well.  It was more for the website to provide the ability to toggle the features that I thought could be interesting.

**Comment by Odonno** [comment]
Oh ok ! You know, the website is not mandatory. If we can do first enable/disable from a single point in the code, it will be great. But, yeah, a website will be the next step. One step at a time. :)

I think the website need to be designed by what you need in Feature Toggle (like toggling by user, by group, by date, ...)

**Comment by gep13** [comment]
Yip, makes sense :+1:
